packages/hydrogen/src/customer/types.ts

```diff
@@ -0,0 +1,137 @@
+import type {
+  ClientReturn,
+  ClientVariablesInRestParams,
+} from '@shopify/hydrogen-codegen';
+
+import type {CrossRuntimeRequest} from '../utils/request';
+
+import type {HydrogenSession} from '../hydrogen';
+
+// Return type of unauthorizedHandler = Return type of loader/action function
+// This type is not exported https://github.com/remix-run/react-router/blob/main/packages/router/utils.ts#L167
+type DataFunctionValue = Response | NonNullable<unknown> | null;
+
+type CustomerAPIResponse<ReturnType> = {
+  data: ReturnType;
+  errors: Array<{
+    message: string;
+    locations?: Array<{line: number; column: number}>;
+    path?: Array<string>;
+    extensions: {code: string};
+  }>;
+  extensions: {
+    cost: {
+      requestQueryCost: number;
+      actualQueryCakes: number;
+      throttleStatus: {
+        maximumAvailable: number;
+        currentAvailable: number;
+        restoreRate: number;
+      };
+    };
+  };
+};
+
+export interface CustomerAccountQueries {
+  // Example of how a generated query type looks like:
+  // '#graphql query q1 {...}': {return: Q1Query; variables: Q1QueryVariables};
+}
+
+export interface CustomerAccountMutations {
+  // Example of how a generated mutation type looks like:
+  // '#graphql mutation m1 {...}': {return: M1Mutation; variables: M1MutationVariables};
+}
+
+export type CustomerClient = {
+  /** Start the OAuth login flow. This function should be called and returned from a Remix action. It redirects the user to a login domain. An optional `redirectPath` parameter defines the final path the user lands on at the end of the oAuth flow. The default redirectPath is the page that ran handleUnauthorized, query or mutate with unauthorized customer, then `/account`. */
+  login: (redirectPath?: string) => Promise<Response>;
```

Comment:
description changed for login

Reply:
> The default redirectPath is the page that ran handleUnauthorized, query or mutate with unauthorized customer, then `/account`.

I'm not sure what this means. It sounds as though, if I click login from `/products/hydrogen`, I will be redirected to `/products/hydrogen/account`. That doesn't seem right.